pandas/lib.pyx

```diff
@@ -321,11 +320,11 @@ def isnullobj(ndarray[object] arr):
     cdef ndarray[uint8_t] result
 
     n = len(arr)
-    result = np.zeros(n, dtype=np.uint8)
+    result = np.empty(n, dtype=np.uint8)
     for i from 0 <= i < n:
         val = arr[i]
-        result[i] = val is NaT or _checknull(val)
-    return result.view(np.bool_)
+        result[i] = _check_all_nulls(val)
+    return result.view(dtype=np.bool_)
 
```

Comment:
you mean the `_check_all_nulls`?  Only object arrays hit this path so I thought it'd be safe to have the most generic version of `checknull` here?

For the `dtype` thing, I forgot to remove it after some experiment, but will remove it in the next update.